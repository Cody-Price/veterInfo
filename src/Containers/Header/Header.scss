@import url('https://fonts.googleapis.com/css?family=Black+Ops+One|Oleo+Script');

.header {
  width: 100%;
  height: 15vh;
  width: 85vw;
  z-index: -1;
  position: fixed;
  left: 15vw;
  background-color: rgba(146, 146, 146, 0.7);
  background-image: url('../../assets/flag.jpg');
  background-size: cover;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  box-shadow: 5px 0px 20px 10px rgba(0, 0, 0, 0.452);
}

.logo-search {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  width: 50vw;
  position: relative;
  top: -20px;
}

h1 {
  font-family: 'Black Ops One', 'Times New Roman', Times, serif;
  font-size: 4vw;
  color: #172353;
  margin-bottom: -12px;
  cursor: pointer;
  width: 50vw;
  text-shadow: 2px 2px 2px white, -2px -2px 2px white, 2px -2px 2px white, -2px 2px 2px white;
}

.red {
  color: rgb(201, 34, 45);
  margin-right: 10px;
}

.search-div {
  width: 100%;
}

.search-input {
  width: 60%;
  border: 2px solid #172353;
  margin: 10px 0 10px 10px;
  height: 20px;
  border-radius: 10px 0 0 10px;
  border-right: solid black 1px;
  box-shadow: 1px 1px 3px white, -1px -1px 3px white;
}

button {
  height: 26px;
  border: 2px solid rgb(201, 34, 45);
  border-left: black solid 1px;
  border-radius: 0 10px 10px 0;
  color: rgb(201, 34, 45);
  position: relative;
  top: -1px;
  width: 80px;
  transition-duration: 0.3s;
  box-shadow: 1px 1px 3px white, -1px -1px 3px white;

  &:hover {
    background-color: rgb(201, 34, 45);
    color: white;
    transition-duration: 0.3s;
    border-color: #172353;
  }

  &:focus {
    outline: none;
  }
}

.grey {
  background-color: rgb(179, 179, 179);
  color: white;

  &:hover {
    background-color: rgb(179, 179, 179);
  }
}

a {
  text-decoration: none;
  color: black;
  cursor: pointer;
}

.flag {
  width: 4vw;
  position: relative;
  top: 10px;
}

.left-flag {
  transform: scaleX(-1);
}

.search-input {
  transition-duration: 0.3s;
  padding-left: 10px;
  
  &:focus {
    transform: scaleY(1.2);
    outline: none;
    transition-duration: 0.3s;
  }

  &:focus + button {
    transform: scale(1.2);
    transition-duration: 0.3s;
  }
}

.search-form button {
  &:active {
    background-color: black;
    color: white;
  }
}

.nav-btn {
  border: 4px solid #172353;
  border-radius: 5px;
  color: #172353;
  width: 80px;
  font-size: 10px;
  position: relative;
  top: 5px;
  height: 30px;
  width: 100px;
  font-weight: bold;

  &:active {
    background-color: black;
    transform: scale(0.9);
    transition-duration: 0s;
  }
}

.nav-left {
  border-radius: 20px 8px 8px 20px;
}

.nav-right {
  border-radius: 8px 20px 20px 8px;
}