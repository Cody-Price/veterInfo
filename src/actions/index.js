export const isLoading = (bool) => {
	return {
		type: 'IS_LOADING',
		isLoading: bool
	}
}

export const addArticlesToStore = (articles) => {
  return {
    type: 'ADD_ARTICLES_TO_STORE',
    articles
  }
}

export const addSourcesToStore = (sources) => {
  return {
    type: 'ADD_SOURCES_TO_STORE',
    sources
  }
}

export const removeArticlesFromStore = () => {
  return {
    type: 'REMOVE_ARTICLES_FROM_STORE'
  }
}