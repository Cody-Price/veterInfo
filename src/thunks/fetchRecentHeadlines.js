import apiKey from '../utilities/apiKey'
import { isLoading, hasErrored } from '../actions/index'
import { cleanArticles } from '../utilities/helper/helper'
import { addArticlesToStore } from '../actions/index'

const fetchRecentHeadlines = (page) => {
  const navPage = page || 1
  const url = `https://newsapi.org/v2/everything?q=+veterans AND military&page=${navPage}&apiKey=${apiKey}`
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const results = await response.json()
      const cleanedArticles = cleanArticles(results)
      dispatch(addArticlesToStore(cleanedArticles))
    } catch(error) {
      dispatch(hasErrored(error.message))
    }
  }
}

export default fetchRecentHeadlines