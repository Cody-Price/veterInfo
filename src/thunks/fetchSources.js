import apiKey from '../utilities/apiKey'
import { isLoading } from '../actions/index'
import { addSourcesToStore } from '../actions/index'

const fetchSources = () => {
  const url = `https://newsapi.org/v2/sources?apiKey=${apiKey}`
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const results = await response.json()
      const sources = results.sources.map(source => {
        return {
          name: source.name,
          id: source.id
        }
      })
      dispatch(addSourcesToStore(sources))
    } catch(error) {
      console.log(error.message)
    }
  }
}

export default fetchSources