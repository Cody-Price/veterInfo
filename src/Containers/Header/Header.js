import React, { Component } from 'react'
import './Header.scss'
import flag from '../../assets/flag-gif.gif'
import { connect } from 'react-redux'
import fetchRecentHeadlines from '../../thunks/fetchRecentHeadlines'
import fetchSearchedHeadlines from '../../thunks/fetchSearchedHeadlines'
import { removeArticlesFromStore } from '../../actions/index'

class Header extends Component {
  constructor() {
    super()
    this.state = {
      search: '',
      pageNumber: 1
    }
  }

  handleSubmit = (e) => {
    e.preventDefault()
    this.props.removeArticlesFromStore()
    this.props.fetchSearchedHeadlines(this.state.search, this.state.pageNumber)
  }

  handleChange = (e) => {
    let {name, value} = e.target;
    this.setState({[name]: value})
  }

  incrementPage = async () => {
    this.props.removeArticlesFromStore()
    let newPage = this.state.pageNumber + 1
    await this.setState({pageNumber: newPage})
    if (this.state.search || this.props.filter) {
      this.props.fetchSearchedHeadlines(this.state.search, this.state.pageNumber, this.props.filter)
    } else {
      this.props.fetchRecentHeadlines(this.state.pageNumber)
    }
  }

  decrementPage = async () => {
    if (this.state.pageNumber > 1) {
      this.props.removeArticlesFromStore()
      let newPage = this.state.pageNumber - 1
      await this.setState({pageNumber: newPage})
      this.props.fetchRecentHeadlines(this.state.pageNumber)
    }
  }

  navToHome = () => {
    this.props.removeArticlesFromStore()
    this.props.fetchRecentHeadlines()
  }

  render() {
    return (
      <header className="header">
        <button className="nav-btn nav-left" onClick={this.decrementPage}>PREVIOUS</button>
        <div className="logo-search">
          <h1 onClick={this.navToHome}><img src={flag} alt="flag" className="flag left-flag" /><span className="red">Veter</span>Informant<img src={flag} alt="flag" className="flag" /></h1>
          <p className="slogan">'Keeping the first ammendment in the hands of those who defended it'</p>
          <div className="search-div">
            <form onSubmit={this.handleSubmit} className="search-form">
              <input
                type="text"
                name="search"
                value={this.state.search}
                onChange={this.handleChange}
                className="search-input"
              />
              <button>Search</button>
            </form>
          </div>
        </div>
        <button className="nav-btn nav-right" onClick={this.incrementPage}>NEXT</button>
      </header>
    )
  }
}

const mapStateToProps = (state) => ({
  filter: state.filter
})

const mapDispatchToProps = (dispatch) => ({
  fetchRecentHeadlines: (page) => dispatch(fetchRecentHeadlines(page)),
  removeArticlesFromStore: () => dispatch(removeArticlesFromStore()),
  fetchSearchedHeadlines: (search, page, filter) => dispatch(fetchSearchedHeadlines(search, page, filter))
})

export default connect(mapStateToProps, mapDispatchToProps)(Header)
