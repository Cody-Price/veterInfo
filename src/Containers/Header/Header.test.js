import Header from './Header'
import React from 'react'
import { shallow } from 'enzyme'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import fetchRecentHeadlines from '../../thunks/fetchRecentHeadlines'
import fetchSources from '../../thunks/fetchSources'
import { mapStateToProps, mapDispatchToProps } from './Header'
import { removeArticlesFromStore } from '../../actions/index'
configure({ adapter: new Adapter() }) 

describe('App', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Header />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe.skip('handleSubmit', () => {
    const preventDefault = jest.fn()
    const mockEvent = {preventDefault}

    wrapper.instance().handleSubmit(mockEvent)

  })

  describe('mapStateToProps', () => {
    it('should return an object', () => {
      const mockState = {
        filter: {branch: 'army'},
        bogusProperty: true
      }
      const expected = {
        filter: {branch: 'army'}
      }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a fetchSearchedHeadlines thunk', () => {
      const mockDispatch = jest.fn()

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchSearchedHeadlines()

      expect(mockDispatch).toHaveBeenCalled()
    })

    it('calls dispatch with a removeArticlesFromStore action', () => {
      const mockArticles = [{name: 'news is good'}, {name: 'news is great'}]
      const mockDispatch = jest.fn()
      const actionToDispatch = removeArticlesFromStore(mockArticles)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.removeArticlesFromStore(mockArticles)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('calls dispatch with a fetchRecentHeadlines thunk', () => {
      const mockDispatch = jest.fn()

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchRecentHeadlines()

      expect(mockDispatch).toHaveBeenCalled()
    })
  })
})