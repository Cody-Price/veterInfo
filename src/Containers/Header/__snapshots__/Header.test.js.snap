// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="grey nav-btn nav-left"
          onClick={[Function]}
        >
          PREVIOUS
        </button>,
        <div
          className="logo-search"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
          >
            <h1
              onClick={[Function]}
            >
              <span
                className="red"
              >
                Veter
              </span>
              Informant
            </h1>
          </NavLink>
          <div
            className="search-div"
          >
            <form
              className="search-form"
              onSubmit={[Function]}
            >
              <input
                className="search-input"
                name="search"
                onChange={[Function]}
                type="text"
                value=""
              />
              <button>
                Search
              </button>
            </form>
          </div>
        </div>,
        <button
          className="nav-btn nav-right"
          onClick={[Function]}
        >
          NEXT
        </button>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "PREVIOUS",
          "className": "grey nav-btn nav-left",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "PREVIOUS",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
            >
              <h1
                onClick={[Function]}
              >
                <span
                  className="red"
                >
                  Veter
                </span>
                Informant
              </h1>
            </NavLink>,
            <div
              className="search-div"
            >
              <form
                className="search-form"
                onSubmit={[Function]}
              >
                <input
                  className="search-input"
                  name="search"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <button>
                  Search
                </button>
              </form>
            </div>,
          ],
          "className": "logo-search",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": <h1
                onClick={[Function]}
              >
                <span
                  className="red"
                >
                  Veter
                </span>
                Informant
              </h1>,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="red"
                  >
                    Veter
                  </span>,
                  "Informant",
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Veter",
                    "className": "red",
                  },
                  "ref": null,
                  "rendered": "Veter",
                  "type": "span",
                },
                "Informant",
              ],
              "type": "h1",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <form
                className="search-form"
                onSubmit={[Function]}
              >
                <input
                  className="search-input"
                  name="search"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <button>
                  Search
                </button>
              </form>,
              "className": "search-div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="search-input"
                    name="search"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />,
                  <button>
                    Search
                  </button>,
                ],
                "className": "search-form",
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "search-input",
                    "name": "search",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Search",
                  },
                  "ref": null,
                  "rendered": "Search",
                  "type": "button",
                },
              ],
              "type": "form",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "NEXT",
          "className": "nav-btn nav-right",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "NEXT",
        "type": "button",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="grey nav-btn nav-left"
            onClick={[Function]}
          >
            PREVIOUS
          </button>,
          <div
            className="logo-search"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
            >
              <h1
                onClick={[Function]}
              >
                <span
                  className="red"
                >
                  Veter
                </span>
                Informant
              </h1>
            </NavLink>
            <div
              className="search-div"
            >
              <form
                className="search-form"
                onSubmit={[Function]}
              >
                <input
                  className="search-input"
                  name="search"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <button>
                  Search
                </button>
              </form>
            </div>
          </div>,
          <button
            className="nav-btn nav-right"
            onClick={[Function]}
          >
            NEXT
          </button>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "PREVIOUS",
            "className": "grey nav-btn nav-left",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "PREVIOUS",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/"
              >
                <h1
                  onClick={[Function]}
                >
                  <span
                    className="red"
                  >
                    Veter
                  </span>
                  Informant
                </h1>
              </NavLink>,
              <div
                className="search-div"
              >
                <form
                  className="search-form"
                  onSubmit={[Function]}
                >
                  <input
                    className="search-input"
                    name="search"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                  <button>
                    Search
                  </button>
                </form>
              </div>,
            ],
            "className": "logo-search",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": <h1
                  onClick={[Function]}
                >
                  <span
                    className="red"
                  >
                    Veter
                  </span>
                  Informant
                </h1>,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="red"
                    >
                      Veter
                    </span>,
                    "Informant",
                  ],
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Veter",
                      "className": "red",
                    },
                    "ref": null,
                    "rendered": "Veter",
                    "type": "span",
                  },
                  "Informant",
                ],
                "type": "h1",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <form
                  className="search-form"
                  onSubmit={[Function]}
                >
                  <input
                    className="search-input"
                    name="search"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                  <button>
                    Search
                  </button>
                </form>,
                "className": "search-div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      className="search-input"
                      name="search"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />,
                    <button>
                      Search
                    </button>,
                  ],
                  "className": "search-form",
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "search-input",
                      "name": "search",
                      "onChange": [Function],
                      "type": "text",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Search",
                    },
                    "ref": null,
                    "rendered": "Search",
                    "type": "button",
                  },
                ],
                "type": "form",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "NEXT",
            "className": "nav-btn nav-right",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "NEXT",
          "type": "button",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
